components/ViewToggle.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+
+interface ViewToggleProps {
+  onClick?: (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
+    state: 'grid' | 'list',
+  ) => void;
+  value?: 'grid' | 'list';
+}
+
+const ViewToggle: React.FC<ViewToggleProps> = (props) => {
+  const { onClick, value } = props;
+
+  const [state, setState] = React.useState(value ?? 'grid');
+
+  const onClickHandler = React.useCallback(
+    (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      const newState = state === 'grid' ? 'list' : 'grid';
+      setState(newState);
+      if (onClick) {
+        onClick(event, newState);
+      }
+    },
+    [state, setState, onClick],
+  );
+
+  let listStyle = '';
+  let gridStyle = '';
+
+  if (state === 'list') {
+    listStyle += ' shadow bg-white';
+  } else {
+    gridStyle += ' shadow bg-white';
+  }
+
+  return (
+    <div
+      className="flex rounded p-1 cursor-pointer bg-gray-500"
+      onClick={onClickHandler}
+    >
+      <div className={`rounded p-1${listStyle}`}>
+        <svg
```

Comment:
Whatever this SVG is, can we make it a React component that we use here for easier reading?